fs-storage/src/file_storage.rs

```diff
@@ -8,51 +9,104 @@ use std::{
     path::{Path, PathBuf},
 };
 
+use crate::base_storage::BaseStorage;
 use data_error::{ArklibError, Result};
 
 const STORAGE_VERSION: i32 = 2;
 const STORAGE_VERSION_PREFIX: &str = "version ";
 
-pub struct FileStorage {
+pub struct FileStorage<K, V>
+where
+    K: serde::Serialize,
+    V: serde::Serialize,
+{
     label: String,
     path: PathBuf,
     timestamp: SystemTime,
+    pub value_by_id: BTreeMap<K, V>,
 }
 
-impl FileStorage {
+impl<K, V> FileStorage<K, V>
+where
+    K: serde::Serialize,
```

Comment:
Generic parameter conditions are generally not mentioned where the struct is declared. This because it needless prevents constructing the struct. It is preferred to mention these conditions when implementing a trait/function that need the conditions.